test/CMakeLists.txt

```diff
@@ -80,6 +86,42 @@ if(GTEST_FOUND)
       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
     )
   endforeach()
-else()
-  message(WARNING "GTest not found. Cannot compile tests!")
 endif()
+
+#######################
+## Python Unit Tests ##
+#######################
+
+if(pybind11_FOUND)
+  # Find Python3 to run tests via ctest
+  find_package(Python3 REQUIRED)
+
+  # Python bindings modules
+  pybind11_add_module(util_caching_py
+    python_bindings.cpp
+  )
+  target_link_libraries(util_caching_py PUBLIC
+    util_caching
+  )
+
+  file(GLOB_RECURSE _py_tests CONFIGURE_DEPENDS "*.py")
+
+  # Copy Python test files to build directory
+  foreach(_py_test ${_py_tests})
+    get_filename_component(_py_test_name ${_py_test} NAME)
+    string(REGEX REPLACE "-test" "" PY_TEST_NAME ${_py_test_name})
+    set(PY_TEST_NAME ${PROJECT_NAME}-pytest-${PY_TEST_NAME})
+
+    message(STATUS
+      "Adding python unittest \"${PY_TEST_NAME}\" with working dir ${PROJECT_SOURCE_DIR}/${TEST_FOLDER} \n _test: ${_py_test}"
+    )
+
+    configure_file(${_py_test} ${PY_TEST_NAME} COPYONLY)
+
+    add_test(NAME ${PY_TEST_NAME}
+      COMMAND ${Python3_EXECUTABLE} ${PY_TEST_NAME}
```

Comment:
I know :sweat_smile: 

But I figured since you need to compile the bindings anyway, this is the easiest way to run the tests (they need to be run inside the build directory, because the compiled bindings are not installed or something like that).